src/main/java/apoc/util/SimpleRateLimiter.java

```diff
@@ -0,0 +1,27 @@
+package apoc.util;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class SimpleRateLimiter {
+
+    private final AtomicInteger countDownLatch = new AtomicInteger(0);
+    private final AtomicLong lastUpdate = new AtomicLong(0);
+
+    private final long timeWindow;
+    private final int operationPerWindow;
+
+    public SimpleRateLimiter(long timeWindow, int operationPerWindow) {
+        this.timeWindow = timeWindow;
+        this.operationPerWindow = operationPerWindow;
+    }
+
+    public boolean canExecute() {
```

Comment:
If I'm not mistaken these are fixed time windows, so that means at the boundary of two windows it can write the full amount on either side?

I was thinking about something like this:
rate = 1/s

on each update we increment a counter
and we also decrement the counter by (now-lastUpdate)*rate
and allow if the counter > 0

which is more of a sliding window.